cli/step_activator.go

```diff
@@ -61,87 +57,25 @@ func (a stepActivator) activateStep(
 		stepYMLPth = activatedStep.StepYMLPath
 		origStepYMLPth = activatedStep.OrigStepYMLPath
 	} else if stepIDData.SteplibSource != "" {
-		stepInfo, didUpdate, err := activateStepLibStep(stepIDData, stepDir, stepYMLPth, isStepLibUpdated, isSteplibOfflineMode)
-		didStepLibUpdate = didUpdate
-
-		stepInfoPtr.ID = stepInfo.ID
-		if stepInfoPtr.Step.Title == nil || *stepInfoPtr.Step.Title == "" {
-			stepInfoPtr.Step.Title = pointers.NewStringPtr(stepInfo.ID)
-		}
-		stepInfoPtr.Version = stepInfo.Version
-		stepInfoPtr.LatestVersion = stepInfo.LatestVersion
-		stepInfoPtr.OriginalVersion = stepInfo.OriginalVersion
-		stepInfoPtr.GroupInfo = stepInfo.GroupInfo
-
+		activatedStep, err := activator.ActivateSteplibRefStep(
+			stepmanLogger,
+			stepIDData,
+			stepDir,
+			workDir,
+			isStepLibUpdated,
+			isSteplibOfflineMode,
+			stepInfoPtr,
+		)
+		didStepLibUpdate = activatedStep.DidStepLibUpdate
 		if err != nil {
-			return "", "", didStepLibUpdate, err
+			return "", "", didStepLibUpdate, fmt.Errorf("activate steplib step reference: %w", err)
```

Comment:
Do we need `reference` at the end of the error message? I think we activate the StepLib step rather than its reference.

Reply:
fixed